packages/react-form-state/src/FormState.ts

```diff
@@ -140,14 +140,14 @@ export default class FormState<
 
   // eslint-disable-next-line @shopify/react-prefer-private-members
   public validateForm() {
-    return new Promise((resolve) => {
+    return new Promise<void>((resolve) => {
```

Comment:
TS now warns if you don't return anything from a promise as it suspects that might be a bug. If that's what you absolutely wanted to do then this is how you say "yes Typescript I really wanted to return nothing from this promise'